File: src/services/api/todos.ts
import { apiRequest } from '@/lib/api-utils';

export interface Todo {
  id: number;
  user_id: string;
  text: string;
  completed: boolean;
  created_at: string;
}

export const todoService = {
  getTodos: async (userId: string): Promise<Todo[]> => {
    return apiRequest<Todo[]>(`/api/todos/${userId}`);
  },

  createTodo: async (userId: string, text: string): Promise<Todo> => {
    return apiRequest<Todo>(`/api/todos/${userId}`, {
      method: 'POST',
      body: JSON.stringify({ text })
    });
  },

  updateTodo: async (todoId: number, completed: boolean): Promise<Todo> => {
    return apiRequest<Todo>(`/api/todos/${todoId}`, {
      method: 'PATCH',
      body: JSON.stringify({ completed })
    });
  },

  deleteTodo: async (todoId: number): Promise<boolean> => {
    const res = await apiRequest<any>(`/api/todos/${todoId}`, {
      method: 'DELETE'
    });
    return res.ok === true;
  }
};

File: backend/api/todos.py
from fastapi import APIRouter, Depends, HTTPException
from sqlmodel import Session, select
from typing import List
from backend.database import get_async_session, AsyncSession
from backend.models.projects import Todo
from pydantic import BaseModel

router = APIRouter(prefix="/api/todos", tags=["Production Queue"])

class TodoCreate(BaseModel):
    text: str

class TodoUpdate(BaseModel):
    completed: bool

@router.get("/{user_id}", response_model=List[Todo])
async def get_todos(user_id: str, session: AsyncSession = Depends(get_async_session)):
    statement = select(Todo).where(Todo.user_id == user_id).order_by(Todo.created_at)
    results = await session.exec(statement)
    return results.all()

@router.post("/{user_id}", response_model=Todo)
async def create_todo(user_id: str, todo_in: TodoCreate, session: AsyncSession = Depends(get_async_session)):
    todo = Todo(user_id=user_id, text=todo_in.text)
    session.add(todo)
    await session.commit()
    await session.refresh(todo)
    return todo

@router.patch("/{todo_id}", response_model=Todo)
async def update_todo(todo_id: int, todo_in: TodoUpdate, session: AsyncSession = Depends(get_async_session)):
    todo = await session.get(Todo, todo_id)
    if not todo:
        raise HTTPException(status_code=404, detail="Todo not found")
    todo.completed = todo_in.completed
    session.add(todo)
    await session.commit()
    await session.refresh(todo)
    return todo

@router.delete("/{todo_id}")
async def delete_todo(todo_id: int, session: AsyncSession = Depends(get_async_session)):
    todo = await session.get(Todo, todo_id)
    if not todo:
        raise HTTPException(status_code=404, detail="Todo not found")
    await session.delete(todo)
    await session.commit()
    return {"ok": True}
